### ISSUE TITLE
Calendar/DateInput Accessibility Audit

### ISSUE THREAD

**Issue by ShimiSun** [grommet/grommet#5325]
Related to #4068 

### Expected Behavior

1. User using a screen reader should be able to tab to the group of days in the calendar and then use the arrow keys to get to the individual days to select a value. If the user hits tab again, they will be taken out of the group of days and into the next element.

2. Next and Previous buttons should indicate to the screen reader user that the view had changed and an action had been performed once the event was triggered and a button was clicked.

### Actual Behavior

When the user tabs to the group of days, the entire calendar and the values of the days are read by the screen reader. The user is able to use the arrow keys to move across days in the group, but the individual value for the day the user is on is not read aloud. 

### Video exhibiting the issue

https://user-images.githubusercontent.com/38971752/122130049-0a6ce880-ce05-11eb-875c-45ef5229c903.mov

### Steps to Reproduce

1. Use VoiceOver for Mac Accessibility (or any other screen reader)
2. Open up the simple story for Calendar in a new window from [Grommet Storybook](https://storybook.grommet.io/?path=/story/visualizations-calendar-simple--simple)

### Your Environment

- Grommet version:  2.17.3
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MackBook Pro

**Comment by ShimiSun** [comment]
@amandadupell

**Comment by amandadupell** [comment]
Related to #4068 

### Expected Behavior

User using a screen reader should be able to tab to the group of days in the calendar and then use the arrow keys to get to the individual days to select a value. If the user hits tab again, they will be taken out of the group of days and into the next element.

### Actual Behavior

When the user tabs to the group of days, the entire calendar and the values of the days are read by the screen reader. The user is able to use the arrow keys to move across days in the group, but the individual value for the day the user is on is not read aloud. 

### Video exhibiting the issue

https://user-images.githubusercontent.com/38971752/122130049-0a6ce880-ce05-11eb-875c-45ef5229c903.mov

### Steps to Reproduce

1. Use VoiceOver for Mac Accessibility (or any other screen reader)
2. Open up the simple story for Calendar in a new window from [Grommet Storybook](https://storybook.grommet.io/?path=/story/visualizations-calendar-simple--simple)

### Your Environment



- Grommet version:  2.17.3
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MackBook Pro

**Comment by amandadupell** [comment]
after looking more into this issue, the current behavior of tabbing to the group of dates and then using the keyboard arrow keys to navigate around in the grid is acceptable. the pain-point now is relaying the information of the active date back to a user via a screen reader so they can know what date they are on. 

some research was done regarding possible solutions to controlling tabbing and other keyboard keys in groups. a similar example to this would be a group of radio buttons or a chart of information where a user does not want to tab through all of the elements in the group. these patterns are called roving index and aria-activedescendant. to learn more about these patterns and when to use them, [here](https://zellwk.com/blog/element-focus-vs-aria-activedescendant/) is a helpful article describing pros and cons of the two.

it's also important to note that roles play a large part in relaying information to a user via screen reader. changes will be made to the calendar to accurately describe the grid layout.
